### DISCUSSION TITLE
REST API GET /repos/{owner}/{repo}/pages 404s

### DISCUSSION THREAD

**Discussion by adriangb** [community/community#24604] [Apps, API and Webhooks]
<p>I’m trying a basic GET to the pages API and am getting 404s. For example:</p>
<pre><code class="lang-bash">curl \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/octocat/hello-world/pages
</code></pre>
<p>Gives me a 404. This is straight from the <a href="https://docs.github.com/en/rest/reference/repos#get-a-github-pages-site" rel="noopener nofollow ugc">docs</a>.</p>
<p>I also tried this with some of my repos, which I know have pages configured, and I’m getting the same thing. Other endpoints (like <code>GET /repos/{owner}/{repo}/branches</code>) work fine.</p>

**Comment by byrneh** [comment, marked as answer]
<p>Hi <a class="mention" href="/adriangb">@adriangb</a><br>
A 404 Not Found from a REST API is often returned due to insufficient permissions (there are security reasons why its a 404, but the explanation is outside the scope of this question)</p>
<p>I was able GET with a Token scoped “repo” (Full control of private repositories) for one of my own Pages.<br>
<a href="https://api.github.com/repos/octocat/hello-world/pages" rel="noopener nofollow ugc">https://api.github.com/repos/octocat/hello-world/pages</a>  returns a 404 Not Found for me also.</p>
<p>Your “gh-pages-docs” is retrievable, when I tried</p>

**Reply by prem-mukherji** [reply]
For me, it was my organization permissions, where it was asking to enable the SSO. I did and it worked.
Thank You.

**Comment by adriangb** [comment]
<p>Thanks for testing <a class="mention" href="/byrneh">@byrneh</a></p>
<p>I can confirm that using a repo-scoped token I am able to get a response.</p>
<p>That begs the question: why is auth required to hit this GET endpoint for public repos, when one can hit many other GET endpoints on public repos (e.g. <code>GET /repos/{owner}/{repo}</code> or <code>GET /repos/{owner}/{repo}/tags</code>) without authenticating. Is there some permissions/security issue I’m not thinking of that requires this behavior? It seems to me that for public repos, any info that is accessible without authenticating on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> should be available without authenticating via API.</p>
<p>Regardless, it seems that <code>octocat/hello-world</code> does not have pages configured, so that should be fixed or the docs should be changed.</p>

**Reply by ReenigneArcher** [reply]
A year later and still seems to be an issue. [Docs](https://docs.github.com/en/rest/pages?apiVersion=2022-11-28#get-a-github-pages-site) were updated, but the new example still doesn't work (with or without the repo scope).

My initial thought was the endpoint only works for repos owned by the token user, but that doesn't seem to be the case as I can get pages info for other public repos, so long as the repo scope is added to the token. The docs still don't make any mention of the token scope for the "GET" method.

Anyway, thanks to your comment I was able to solve my issue.

**Reply by monikacp** [reply]
How you solved this issue? I am still facing 404 error.

**Reply by ReenigneArcher** [reply]
Add "repo" scope to your PAT, and make sure you authenticate.

**Reply by Krushit-Babariya** [reply]
still i am not able to solve 404.

**Comment by byrneh** [comment]
<p><a class="mention" href="/adriangb">@adriangb</a>, I guess it could equally be argued why would authorization not be required, I am not defending one position or the other on this matter…<br>
I did notice the API returns the details of GitHub pages and these can be public or private, so an authentication header does not some unreasonable to determine whether the requestor is authorised to receive the result of a Private GitHub Page.</p>
<p>If you are happy your original question is now solved, you can mark as resolved.</p>

**Comment by adriangb** [comment]
<p>Thank you for the answer. I’ll mark your answer as a solution, although I still think that the following should be done</p>
<ol>
<li>The documentation or example repo are fixed.</li>
<li>A clear explanation is given regarding the permissions for this endpoint.</li>
</ol>

**Comment by byrneh** [comment]
<p>I would agree in some cases the documentation does do not detail a lot of extra information other than the fields and whether it needs to be authenticated or not, so the addition of minimum scope permissions is useful to avoid users having to work out for themselves and also not get confused that a 404 Not Found may be the result of incorrectly scoped permissions.<br>
I would also agree it is less confusing for a user reading docs if any example shown works.</p>

**Comment by apertureatf1** [comment]
<p>Hi, Byrneh<br>
First, I have a similar issue. I am able to read repos by using a token.</p>
<p>Second, I am also able to add collaborators use the same token</p>
<p><a href="https://api.github.com/repos/apertureqa/new_repo/collaborators/apertureatf1" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/apertureqa/new_repo/collaborators/apertureatf1</a></p>
<p>however, when I try to upload a file to the same repo. I got not found error error code 404. I guess it is related with privilege? what should I do manually to change the file test priviliage to allow additional content can be add to test. I manuall created the following file in the repo. what kinds of permission I need to manually update on this file to allow the API call PUT go though to add content to the file? thank you very much</p>
<p>I have been blocked, it will be</p>
<p>‘<a href="https://api.github.com/repos/apertureqa/new_repo/test" rel="noopener nofollow ugc">https://api.github.com/repos/apertureqa/new_repo/test</a>’</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/apertureatf1">@apertureatf1</a>,<br>
This is not really a similar issue, It is best practise and helps all involved greatly with readability and understanding and for others to support you by creating new issues/questions specific to your problems, rather than various different threads of one issues, it gets messy and confusing.<br>
Don’t worry people in the community will still help resolve any issues in your new thread.<br>
you can also mention a user with an '@'followed by their name.</p>
<p>Your question is not altogether clear, but if I was to guess it sounds like you are trying to use the REST api incorrectly, user will work with git client/GUI’s for updating content.<br>
Any action possible through the REST api is documented here <a href="https://docs.github.com/en/rest" rel="noopener nofollow ugc">GitHub REST API - GitHub Docs</a>.<br>
If you are a new year the simplest thing to do is follow one of the quick guided tutorials.<br>
If you head over to GitHub’s <a href="https://lab.github.com/" rel="noopener nofollow ugc">GitHub Learning Lab</a> and  Introduction to GitHub’ should get you started using GitHub in less than an hour.<br>
I have found a lot of occasional users now using GitHub Desktop or similar GUIs see <a href="https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/getting-started-with-github-desktop" rel="noopener nofollow ugc">Getting started with GitHub Desktop - GitHub Docs</a>, it avoids the need to know and remember a lot of command line syntax etc.</p>
<p>I also tried the REST API method for <a href="https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents" rel="noopener nofollow ugc">create-or-update-file-contents</a>  if that is what you were attempting</p>
<ul>
<li>accidentally using GET rather than PUT IN HTTP verb returns status 404 Not Found</li>
<li>content Base64 encoded and using PUT I got status 422 “message”: “content is not valid Base64”,</li>
<li>The same content again but converted using with Base64 with padding and it worked status 201<br>
I was able to update content with PUT and delete with DELETE</li>
</ul>
<p>Good luck</p>

**Comment by fceller** [comment]
<p>I have a token that has all “repo” entries, but I’m still getting 404. Any idea what could be this issue in this case</p>
<pre><code class="lang-auto">&gt; curl -X GET -H "Accept: application/vnd.github.v3+json" -H "Authorization: $GITHUB_TOKEN" https://api.github.com/repos/arangodb/justfortesting/pages
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-github-pages-site"
}
</code></pre>

**Comment by brooookemiller** [comment]
I was facing the same issue and updated my personal access token to include all of the scopes selected under "repo", which did the trick for me. I previously only had selected:

-  repo:status Access commit status
-  public_repo Access public repositories

but it seems that the other repo scope selections were needed. I would **highly** suggest clarifying this scope need in the docs.

**Reply by dpuga1** [reply]
Repo

**Reply by guenan01** [reply]
This was my solution. I was trying to use fine grained tokens at first but failing. I had to use classic tokens, include ALL repo checkboxes, and then Configure SSO once the token was generated. I don't see an obvious way to configure SSO authorization with fine grained tokens.

**Comment by aeonian-joy** [comment]
I'm facing 404 Not Found messages while creating a repo using postman. I have an PAT and the same I am using for GET request, and it is giving me 200 OK but same is failing while POST method. Why so?

**Comment by dpuga1** [comment]
I don’t know I really new to this
David Puga

**Comment by riverdusty** [comment]
This is still a problem. [Docs ](https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28#repository-permissions-for-issues )say "no additional permissions". New BETA token creator set to 'All repositories' and 'Repository permissions = issues = read only access'... Generated token. Still get 404 on GET repos/USER/REPO/issues. (The repo is a private repo)

![image](https://github.com/community/community/assets/572263/b09bcb91-8394-4e95-944c-d56d323e4248)
![image](https://github.com/community/community/assets/572263/920196f7-175f-4419-98eb-524fa95e959f)
![image](https://github.com/community/community/assets/572263/cf42b951-8fc5-4a48-8379-d7cb81383792)

Its clear, if I'm doing something wrong, its because its not covered in the documentation.

**Reply by danhdox** [reply]
any update to this? im hitting the same issue

**Comment by boctulus** [comment]
Same problem here in 2024 with different kind of tokens.

**Comment by monikacp** [comment]
Thank you. It's working for me.

Best Regards,

Monika Sahu

DevOps Engineer

c. +91 8827305731

e. ***@***.***

Cloudphysician Healthcare Private Limited

7, Bellary Road, Ganganagar

Bengaluru, 560032, KA, India.

cloudphysician.net  <http://www.cloudphysician.net>

**Comment by lxRbckl** [comment]
Hey!
I also ran into this issue! Changing permissions was the right directions like the solution says, but in some cases the problem can still persist, like in my case. I ended up fixing this issue by deleting the token entirely and creating a new token with those same permissions. That fixed the issue for me.
